### PR TITLE
Use Source.queue

### PR DESCRIPTION
**Pull request by laglangyue**
OverflowStrategy.dropNew has been deprecated
Use Source.queue

### CODE REVIEW

File: docs/src/test/scala/docs/http/scaladsl/HttpClientExampleSpec.scala [hunk 1]
```diff
@@ -13,31 +13,27 @@
...
 
-import scala.concurrent.ExecutionContext
-import org.apache.pekko
-import pekko.http.scaladsl.model.HttpRequest
-import pekko.http.scaladsl.settings.ClientConnectionSettings
-import pekko.http.scaladsl.settings.ConnectionPoolSettings
-import scala.annotation.nowarn
 import docs.CompileOnlySpec
+import org.apache.pekko
 import org.scalatest.matchers.should.Matchers
```
Comment (pjfanning) [inline docs/src/test/scala/docs/http/scaladsl/HttpClientExampleSpec.scala:18]: Keep this with the other pekko imports

File: docs/src/test/scala/docs/http/scaladsl/Http2ClientApp.scala [hunk 2]
```diff
@@ -75,28 +74,31 @@ object Http2ClientApp extends App {
...
 
-  // OverflowStrategy.dropNew has been deprecated in latest Pekko versions
-  // FIXME: replace with 2.6 queue when 2.5 support is dropped, see #3069
-  @nowarn("msg=Use Source.queue") //
   // #response-future-association
   def singleRequest(
       connection: Flow[HttpRequest, HttpResponse, Any], bufferSize: Int = 100): HttpRequest => Future[HttpResponse] = {
-    val queue =
-      Source.queue(bufferSize, OverflowStrategy.dropNew)
+    val queue = {
```
Comment (He-Pin) [inline docs/src/test/scala/docs/http/scaladsl/Http2ClientApp.scala:80]: The {} is not needed

File: docs/src/test/scala/docs/http/scaladsl/HttpClientExampleSpec.scala (hunk 1 above)
Comment (He-Pin) [reply docs/src/test/scala/docs/http/scaladsl/HttpClientExampleSpec.scala:18]: Yes, sort it with idea or sortimports

**Comment by He-Pin** [comment]
Cool, thanks, I will check it locally after Friday work

File: docs/src/test/scala/docs/http/scaladsl/Http2ClientApp.scala [hunk 3]
```diff
@@ -21,13 +21,12 @@ import pekko.http.scaladsl.model.headers.HttpEncodings
 import pekko.http.scaladsl.model.HttpRequest
 import pekko.http.scaladsl.model.HttpResponse
 import pekko.http.scaladsl.model.headers
-import pekko.stream.OverflowStrategy
 import pekko.stream.scaladsl.Flow
 import pekko.stream.scaladsl.Sink
 import pekko.stream.scaladsl.Source
 import com.typesafe.config.ConfigFactory
+import org.apache.pekko.stream.QueueOfferResult
```
Comment (samueleresca) [inline docs/src/test/scala/docs/http/scaladsl/Http2ClientApp.scala:28]: `pekko.stream.QueueOfferResult`?

**Comment by laglangyue** [comment]
thanks for all @pjfanning @He-Pin @samueleresca . I used an idea sort and further manually adjusted it, following the rules of pekko - empty line - and then third-party libraries.

I can't use sortImports in pekko-http, because we not add plugin scalafix

**Review by pjfanning — APPROVED** [review]
lgtm

File: docs/src/test/scala/docs/http/scaladsl/Http2ClientApp.scala [hunk 4]
```diff
@@ -94,9 +91,14 @@ object Http2ClientApp extends App {
...
       val p = Promise[HttpResponse]()
-      queue.offer(req.addAttribute(ResponsePromise.Key, ResponsePromise(p)))
+      queue.offer(req.addAttribute(ResponsePromise.Key, ResponsePromise(p))) match {
         // return the future response
-        .flatMap(_ => p.future)
+        case QueueOfferResult.Enqueued    => p.future
+        case QueueOfferResult.Dropped     => Future.failed(new RuntimeException("Queue overflowed. Try again later."))
+        case QueueOfferResult.Failure(ex) => Future.failed(ex)
+        case QueueOfferResult.QueueClosed => Future.failed(
+            new RuntimeException("Queue was closed (pool shut down) while running the request. Try again later."))
```
Comment (He-Pin) [inline docs/src/test/scala/docs/http/scaladsl/Http2ClientApp.scala:100]: should be `submitting the request`

**Review by He-Pin — CHANGES_REQUESTED** [review]
overall lgtm, please update the exception msg

File: docs/src/test/scala/docs/http/scaladsl/Http2ClientApp.scala [hunk 5]
```diff
@@ -97,7 +97,7 @@ object Http2ClientApp extends App {
         case QueueOfferResult.Dropped     => Future.failed(new RuntimeException("Queue overflowed. Try again later."))
         case QueueOfferResult.Failure(ex) => Future.failed(ex)
         case QueueOfferResult.QueueClosed => Future.failed(
-            new RuntimeException("Queue was closed (pool shut down) while running the request. Try again later."))
+            new RuntimeException("Queue was closed (pool shut down) while running the request."))
```
Comment (He-Pin) [inline docs/src/test/scala/docs/http/scaladsl/Http2ClientApp.scala:100]: it's not `running`, just submitting or attemptling

**Review by He-Pin — APPROVED** [review]
lgtm, thanks

**Pull request merged**